Every day, optionally handoff sequencer lockout to compact database

PlasmaPower opened the pull request:
I've split this PR up into two commits: One that refactors how we do the handoff on shutdown, and one that uses that refactor to also do a regular handoff to compact the database. I'll probably make more commits in response to the PR review, but I think splitting it up should make review easier.

tsahee reviewed (commented):
I really like the logic (especially with new naming). A few small-ish comments

PlasmaPower commented on arbnode/seq_coordinator.go:
```diff
@@ -603,24 +629,24 @@ func (c *SeqCoordinator) update(ctx context.Context) time.Duration {
-		livelinessErr = c.livelinessUpdate(ctx)
-		if livelinessErr == nil {
-			c.reportedAlive = true
+		wantsLockoutErr = c.wantsLockoutUpdate(ctx)
```
I've gone with option 2 :) I've renamed chosenUpdateMutex to wantsLockoutMutex and acquired it in wantsLockoutUpdate. Then, I checked for that variable directly in wantsLockoutRelease, which means that the variable isn't needed outside of the functions that already have the mutex and are modifying the wants lockout status.

PlasmaPower commented on arbnode/seq_coordinator.go:
```diff
@@ -653,6 +645,8 @@ func (c *SeqCoordinator) update(ctx context.Context) time.Duration {
 }
 
 func (c *SeqCoordinator) DebugPrint() string {
+	c.wantsLockoutMutex.Lock()
```
I've added a warning that it acquires the mutex, but generally I think it's fine to acquire the mutex briefly to read fields

PlasmaPower commented:
For whatever reason I can't respond directly to your middle two comments in your last PR review, but I've renamed "chosen one" methods and added a check to release the wants lockout key if we're avoiding the lockout.

tsahee reviewed (approved):
LGTM
